serial-central.js

```diff
@@ -0,0 +1,94 @@
+var noble = require('noble');
+var readline = require('readline');
+
+var serviceUUID = 'F9BB83582069437EA3D70BF0E1870F7D'.toLowerCase();
+var characteristicUUID = '30D4E7E20CCE40FEB5DD6565D9BBFF8B'.toLowerCase();
+
+var serialPeripheral;
+var serialCharacteristic;
+
+var logError = function(error) {
+  if (error) {
+    console.log('error: ' + error);
+  }
+};
+
+noble.on('stateChange', function(state) {
+  console.log('on -> stateChange: ' + state);
+
+  if (state === 'poweredOn') {
+    //start scanning for the service we're interested in, not allowing duplicates
+    console.log('Begin scanning for service: ' +  serviceUUID);
+    noble.startScanning([serviceUUID]);
+    // noble.startScanning();
+  } else {
+    noble.stopScanning();
+  }
+});
+
+noble.on('discover', function(peripheral) {
+  console.log('peripheral discovered (' + peripheral.uuid+ '):');
+  console.log('\thello my local name is:');
+  console.log('\t\t' + peripheral.advertisement.localName);
+  console.log('\tcan I interest you in any of the following advertised services:');
+  console.log('\t\t' + JSON.stringify(peripheral.advertisement.serviceUuids));
+
+  if (peripheral.advertisement.serviceUuids.indexOf(serviceUUID) == -1) {
+    console.log('the discovered peripheral does not have the required service.');
+    process.exit(0);
+  }
+
+  //save the peripheral for later
+  serialPeripheral = peripheral;
+  peripheral.connect(function(error) {
```

Comment:
Here you start off a call that then needs to call a couple more things in order. To avoid the nested callback approach there is a pretty popular lib called [async](https://github.com/caolan/async). In particular, check out these control flow items: [series](https://github.com/caolan/async#seriestasks-callback) & [waterfall](https://github.com/caolan/async#waterfall). I've got plenty of examples in some of my side projects so let me know if you want to see some.